Uplift/Views/FacilitiesDropdownHeaderView.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  FacilitiesDropdownHeaderView.swift
+//  Alamofire
+//
+//  Created by Cameron Hamidi on 11/26/19.
+//
+
+import UIKit
+
+class FacilitiesDropdownHeaderView: DropdownHeaderView {
+
+    private enum FacilityName: String {
+        case bowlingLanes = "BOWLING LANES"
+        case equipment = "EQUIPMENT"
+        case fitnessCenter = "FITNESS CENTER"
+        case gymnasium = "GYMNASIUM"
+        case swimmingPool = "SWIMMING POOL"
+    }
+
+    private let headerImageView = UIImageView()
+    private let headerNameLabel = UILabel()
+    private let headerOpenLabel = UILabel()
+
+    init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func configure(for facility: Facility) {
+        var headerImage: UIImage?
+        switch FacilityName(rawValue: facility.name.uppercased()) {
+        case .equipment, .fitnessCenter:
+            headerImage = UIImage(named: ImageNames.equipment)
+        case .gymnasium:
+            headerImage = UIImage(named: ImageNames.basketball)
+        case .swimmingPool:
+            headerImage = UIImage(named: ImageNames.pool)
+        case .bowlingLanes:
+            headerImage = UIImage(named: ImageNames.bowling)
+        default:
+            headerImage = UIImage(named: ImageNames.misc)
+        }
+
+        headerImageView.image = headerImage
+        addSubview(headerImageView)
+
+        headerNameLabel.text = facility.name
+        headerNameLabel.font = ._16MontserratRegular
+        headerNameLabel.textColor = .black
+        headerNameLabel.textAlignment = .left
+        addSubview(headerNameLabel)
+
+        headerOpenLabel.font = ._16MontserratRegular
+        headerOpenLabel.textAlignment = .center
+        addSubview(headerOpenLabel)
+
+        if let isFacilityOpen = getIsFacilityOpen(for: facility) {
+            headerOpenLabel.text = isFacilityOpen ? ClientStrings.CommonStrings.open : ClientStrings.CommonStrings.closed
+            headerOpenLabel.textColor = isFacilityOpen ? .accentOpen : .accentClosed
+        }
+
+        setupConstraints()
+    }
+
+    func getIsFacilityOpen(for facility: Facility) -> Bool? {
+        let todaysDate = Date()
+        let dayOfWeek = todaysDate.getIntegerDayOfWeekToday()
+        for detail in facility.details {
+            if !detail.times.isEmpty {
+                for time in detail.times {
+                    if time.dayOfWeek == dayOfWeek {
+                        for timeRange in time.timeRanges {
+                            if todaysDate > timeRange.openTime && todaysDate < timeRange.closeTime {
+                                return true
+                            } else {
+                                return false
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return nil
```

Comment:
what does it mean if we return `nil` here? do we handle it same way we handle `false`?

Reply:
If the facility in question is something that doesn't have hours associated with it and isn't something that can be "open" or "closed" (ie free weights or miscellaneous information), then nil is returned and the header view doesn't set an open/closed label.

Reply:
May be worth creating an enum to document this in code, or writing actual documentation. I was confused reading this block as well.